### ISSUE TITLE
IText style lost in new line

### ISSUE THREAD

**Issue by hellomaya** [kangax/fabric.js#2169]
I didn't find any feature like char wrap or word wrap of IText or Text from [fabricjs.com/docs](http://fabricjs.com/docs), and I am trying to make that, while exchanging HTML and plain Text between Canvas and an online editor in fabricjs (I)Text objects.

I saw IText had ```setSelectionStart```, ```setSelectionEnd``` , ```setSelectionStyle```, which is pretty cool, and there comes a problem, when the text wrapped, the style will be lose for new line, for example, if I have a one line text "Hello world, FabricJs is cool", and it had a red color and font size style, then for wrap reason, I will change it to  "Hello world,\n FabricJs is cool", then I will re-draw it with same IText object, I found that it kept the style of "Hello world,", but lose styles for " FabricJs is cool".

I haven't reviewed code in fabricjs yet, I guess the styles are re-calculated based on different lines, but should it keep the styles?

That happened in fabricjs 1.4.13

**Comment by asturur** [comment]
word wrap is coming with a different PR, number #1778 ( textbox class )

Style problems on new lines should be solved, could you please try with the latest code base?

how do you trigger the bug? using mouse and keyboard or using the code?

**Comment by hellomaya** [comment]
Good to know there is a textbox class, and I did the same thing with char/word wrap, I work around current IText limitation by wrapping text & rectangle to group and finished it, the styles lost when I tried to insert more "\n" into the original text string. A native textbox is more than welcome! Just a few idea, actually IText is cool, but not really useful for printing app, because in printing app, we need calculate the size of IText in metrics system, without word wrap or char wrap, it's almost useless, that's the critical feature. Sometimes It doesn't matter if we can edit text directly or not...
